keps/0014-pull-request-process.md

```diff
@@ -0,0 +1,129 @@
+   * [Outside contributor](#outside-contributor)
+   * [Trusted contributor](#trusted-contributor)
+   * [Repository administrator](#repository-administrator)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+In order to ensure that contributors to the KUDO project have a good experience, we have a streamlined pull request review and merging process using Prow.
+
+## Goals
+
+* Automate and enforce labeling of pull requests and issues (e.g., size).
+* Automate pull request and issue life cycles.
+* Automatically request pull request review from the correct people.
+* Allow contributors to restrict who is allowed to run tests.
+* Not be limited by public CI services (queue times, concurrency, resources).
+* Use CNCF-standard tooling for our processes.
+
+## Proposal
+
+A Prow cluster will be deployed in Kubernetes for running CI and PR/issue automation. This cluster lives at https://prow.kudo.dev/ and will take action on Github as the `ci-mergebot` user. KEP-0004 outlines the infrastructure required.
+
+### Issues and pull requests
+
+Prow will be used to manage issue and pull request labels and life cycle. When an issue or pull request is created or commented on, it will be scanned for Prow commands (lines that begin with `/`).
+
+Supported commands:
+
+* `/help`: labels the issue or PR as wanting help from external contributors.
+* `/good-first-issue`: labels the issue or PR as a good issue for new contributors to tackle.
+* `/priority`: sets the priority on the issue or PR.
+* `/milestone`: set a milestone for the issue or PR.
+* `/lifecycle`: set a specific lifecycle label on the issue or PR.
+* `/kind`: label the issue or PR as a specific kind (e.g., bug, enhancement, etc).
+* `/close`: close an issue or PR .
+* `/repoen`: open an issue on PR that has been closed.
+
+When issues and PRs are 30 days old, they will be labeled `lifecycle/rotten` and closed automatically at 60.
+
+### Pull requests
+
+For pull requests only, Prow will automatically request review from the proper users, control when tests are run (and run them), report test status, and automatically merge the pull request when tests are passed and it has the proper approvals.
+
+Supported commands:
+
+* `/ok-to-test`: if the PR creator is an external contributor, Prow will not run tests on the pull request until a contributor has specified `/ok-to-test`.
+* `/test`: re-run all of the tests or a specific test.
+* `/retest`: re-run failed tests.
+* `/override`: ignore a test's status.
+* `/approve`: approve a pull request.
+* `/lgtm`: mark a pull request as ready for merge, once it has the required number of approvals and all tests are passed, Prow will merge it automatically.
+* `/cc`: request review from a specific user.
+* `/hold`: prevent a pull request from being merged.
+
+Once two contributors have marked `/approve` (or `/lgtm`, which implies `/approve`) and one types `/lgtm` and all tests pass, a pull request will be automatically merged.
```

Comment:
1. I don't get the diff between approve and lgtm... they seem basically the same... why have 2?  generally when using GH people will "approve" with "lgtm" comment.    It isn't clear what we are trying to achieve.

Reply:
Regarding this sentence.  It seems implied (which we should make explicit) that the PR does not have a merge conflict.

Reply:
Good question, I'll have to do some digging to see what the rationale is for it to work that way. My guess would be that allows someone to differentiate between "I think this is good code" and "I think this is ready to be merged as soon as the tests pass" - not all approvers may want to pull that trigger.

Reply:
I'm not finding their rationale, but this is configurable so if we'd rather just have to /approve we can do that instead.